client/src/App.js

```diff
@@ -1,11 +1,175 @@
-import "./App.css";
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
+import InputGroup from "react-bootstrap/InputGroup";
+import Container from "react-bootstrap/Container";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import Video from "./Video";
 
 function App() {
+  const [title, setTitle] = useState("");
+  const [url, setUrl] = useState("");
+  const [items, setItems] = useState(null);
+  const [search, setSearch] = useState("");
+  const [isOpen, setIsOpen] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isAscending, setIsAscending] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await axios.get(
+          `https://full-stack-assessment.onrender.com/videos?ordering=${
+            isAscending ? "asc" : "desc"
+          }`
+        );
+        setIsLoading(true);
+        setItems(response.data);
+      } catch (e) {
+        console.log(e);
+      }
+    };
+
+    fetchData();
+  }, [isAscending]);
+
+  const orderHandler = (e) => {
+    setIsAscending(!isAscending);
+  };
+
+  const searchHandler = async () => {
+    const res = await axios.get(
+      `https://full-stack-assessment.onrender.com/search/?title=${search}`
+    );
+    setItems(res.data);
+  };
+
+  const openHandler = () => {
+    setIsOpen(true);
+  };
+
+  const cancelHandler = () => {
+    setIsOpen(false);
+  };
+
+  const urlHandler = (event) => {
+    setUrl(event.target.value);
+  };
+
+  const titleHandler = (event) => {
+    setTitle(event.target.value);
+  };
+
+  const validateUrl = (value) => {
+    // Regular expression for URL validation
+    const urlRegex = /^(ftp|http|https):\/\/[^ "]+$/;
+    if (!value || !urlRegex.test(value)) {
+      alert("Please enter a valid URL");
+    }
+    return null; // Return null for no validation errors
+  };
+
+  const addClickHandler = async () => {
+    if (!title) {
+      alert("Please enter a title");
+      return;
+    }
+    validateUrl(url);
+    try {
+      const response = await axios.post(
+        "https://full-stack-assessment.onrender.com/video",
+        {
+          title: title.toLowerCase(),
+          url: url.split("watch?v=")[1],
+        }
+      );
+      setTitle("");
+      setUrl("");
+      alert("Video has been sent");
+      window.location.reload();
```

Comment:
Ideally you'll use the response from the server to update the videos state so the videos list will re-render without needing a full page load.